.circleci/config.yml

```diff
@@ -18,117 +18,135 @@ commands:
   detect_root_change:
     description: Detects root file changes
     steps:
-      - run: |
-          set -e
-          trap 'catch $? $LINENO' ERR
+      - run:
+          name: Detect root file changes (if so, treat all packages as updated)
+          command: |
+            set -e
+            trap 'catch $? $LINENO' ERR
 
-          catch() {
-            if [ "$1" != "0" ]; then
-              # error handling goes here
-              echo "Error $1 occurred on $2"
-            fi
-          }
+            catch() {
+              if [ "$1" != "0" ]; then
+                # error handling goes here
+                echo "Error $1 occurred on $2"
+              fi
+            }
 
-          echo 'Checking changes'
-          changes=`git diff --name-only origin/master...$CIRCLE_BRANCH | { grep -Ev '^packages/|yarn.lock|bear.png|.editorconfig' || true; }`
-          echo 'Changes detected:'
-          echo $changes
+            echo 'Checking changes'
+            changes=`git diff --name-only origin/master...$CIRCLE_BRANCH | { grep -Ev '^packages/|yarn.lock|bear.png|.editorconfig' || true; }`
+            echo 'Changes detected:'
+            echo $changes
 
-          if [[ $changes ]] || [ $CIRCLE_BRANCH == "master" ]; then
-            echo 'export RUN_ALL=true' >> $BASH_ENV
-            source $BASH_ENV
-          else
-          echo 'export RUN_ALL=false' >> $BASH_ENV
-            echo 'Root files not changed'
-          fi
+            if [[ $changes ]] || [ $CIRCLE_BRANCH == "master" ]; then
+              echo 'export RUN_ALL=true' >> $BASH_ENV
+              source $BASH_ENV
+            else
+            echo 'export RUN_ALL=false' >> $BASH_ENV
+              echo 'Root files not changed'
+            fi
 
-  echo_root:
-    description: Detects root package.json changes
-    steps:
-      - run: |
-          echo $RUN_ALL
+  # echo_root:
```

Comment:
This step doesn't seem to do anything yet but I've left it in (but commented out) so it can be implemented later.

Reply:
Will need to check with @siggerzz but this may be linked to the checks that run if a root change is found.

It equally might just echo out the variable being used (which is what it looks like), so although not crucial to the actual build, may be something we want for debug purposes?